biblib/tests/unit_tests/test_views.py

```diff
@@ -1857,6 +1857,219 @@ def test_cannot_update_libraries_details_if_not_owner_or_admin(self):
+    """
+    Base class to test the Document view for POST/DELETE (PUT for tags?)
+    """
+
+    def __init__(self, *args, **kwargs):
+        """
+        Constructor of the class
+
+        :param args: to pass on to the super class
+        :param kwargs: to pass on to the super class
+
+        :return: no return
+        """
+
+        super(TestQueryViews, self).__init__(*args, **kwargs)
+        self.query_view = QueryView
+        self.document_view = DocumentView
+        # Stub data
+        self.stub_user = self.stub_user_1 = UserShop()
+        self.stub_user_2 = UserShop()
+
+        self.stub_library = self.stub_library_1 = LibraryShop()
+        self.stub_library_2 = LibraryShop()
+
+    def test_user_can_add_to_library(self):
+        """
+        Tests that adding a bibcode to a library works correctly
+
+        :return:
+        """
+
+        # Ensure a user exists
+        user = User(absolute_uid=self.stub_user.absolute_uid)
+        with self.app.session_scope() as session:
+            session.add(user)
+            session.commit()
+
+            # Ensure a library exists
+            library = Library(name='MyLibrary',
+                              description='My library',
+                              public=True)
+
+            # Give the user and library permissions
+            permission = Permissions(permissions={'read': True, 'write': True, 'admin': False, 'owner': False})
+
+            # Commit the stub data
+            user.permissions.append(permission)
+            library.permissions.append(permission)
+            session.add_all([library, permission, user])
+            session.commit()
+
+            library_id = library.id
+
+            # Get stub data for the document
+
+            # Add a document to the library
+            with MockSolrQueryService(canonical_bibcode = self.stub_library.document_view_post_data('add').get('bibcode')) as SQ:
+                number_added = self.query_view.add_document_to_library(
+                    library_id=library_id,
+                    document_data=self.stub_library.query_view_post_data()
+                )
+            self.assertEqual(number_added.get("number_added"), len(self.stub_library.bibcode))
+
+            # Check that the document is in the library
+            library = session.query(Library).filter(Library.id == library_id).all()
+            for _lib in library:
+                self.assertIn(list(self.stub_library.bibcode.keys())[0], _lib.bibcode)
+
+            # Add a different document to the library
+            with MockSolrQueryService(canonical_bibcode = self.stub_library_2.document_view_post_data('add').get('bibcode')) as SQ:
+                number_added = self.query_view.add_document_to_library(
+                    library_id=library_id,
+                    document_data=self.stub_library_2.query_view_post_data()
+                )
+            self.assertEqual(number_added.get("number_added"), len(self.stub_library.bibcode))
```

Comment:
Shouldn't the second part of this assert be `len(self.stub_library_2.bibcode)`, not the same but for `self.stub_library`?